controllers/storageoscluster/node-manager_operand.go

```diff
@@ -303,11 +308,29 @@ func getNodeManagerBuilder(fs filesys.FileSystem, obj client.Object, kcl kubectl
 		images = append(images, *img)
 	}
 
+	mutators := []kustomize.MutateFunc{
+		kustomize.AddNamespace(cluster.GetNamespace()),
+		kustomize.AddImages(images),
+	}
+
+	sidecars := 0
+
+	// Append upgrade guard as sidecar.
+	if _, ok := cluster.Spec.NodeManagerFeatures[upgradeGuardFeatureKey]; ok {
+		image := cluster.Spec.Images.UpgradeGuardContainer
+		if image == "" {
+			image = os.Getenv(kubeUpgradeGuardEnvVar)
+		}
+		if image != "" {
+			mutators = append(mutators, func(k *kustomizetypes.Kustomization) {
+				sidecars += 1
+				k.Patches = append(k.Patches, generateSideCarContainer(sidecars, "upgrade-guard", image))
+			})
+		}
+	}
```

Comment:
Should this be a separate function that is generic to any sidecar container? (for future-proofing)